### ISSUE TITLE
Data dimension for RNN LSTM??

### ISSUE THREAD

**Issue by RogerBorras** [rstudio/keras#332]
Hi,

I'm training a model from an heart ECG data. With the follwing code:

`library(keras)`

`setwd("D:/SESSIO_DEEP_LEARNING/BEAMER")`

`AF <- read.csv("AF_mod.txt")`
`AF$diag <- 1`
`SR <- read.csv("SR_mod.txt")`
`SR$diag <- 0`

`dades <- rbind(AF, SR)`

`dades_train <- dades[c((1:5000),(120001:125000)), ]`

`X_train <- dades_train[,-13]`
`dim(X_train)`

`[1] 10000    12`

`Y_train <- dades_train[,13]`
`length(Y_train)`
`[1] 10000`

`model <- keras_model_sequential() %>%`
 ` layer_lstm(units = 32, return_sequences = TRUE, input_shape = c(?, ?)) %>%`
`  layer_lstm(units = 32, return_sequences = TRUE) %>%`
`  layer_lstm(units = 32) %>% `
`  layer_dense(units = 2, activation = "sigmoid")`

` model %>% fit(X_train, Y_train, epochs = 5, batch_size = 128) `

` model %>% compile(optimizer = "rmsprop", loss = "binary_crossentropy")`

Where each row is an observation, with 12 features by row. Which data dimension I need, and wich input shape?

Thanks for this great package!

**Comment by remya-kannan** [comment]
@RogerBorras , input_shape has a format: input_shape = c(batch_size, timesteps, data_dim). Keras follows the convention that the batch_size is omitted in the input_shape argument during model building. Batch_size is later called during the fit process. So, going by the definition your input_shape = c(timesteps, dim(X_train)[[2]]). Just a tip, you may want to consider reshaping your array as I foresee you might run into some problems in the future. Hope this helps!

**Comment by RogerBorras** [comment]
Yes, helps a lot. Thanks!! :)
